shed/event_streams.py

```diff
@@ -251,18 +251,55 @@ def curate_streams(self, nds):
         If we get multiple streams make (name, (doc, doc, doc, ...))
         Otherwise (name, (doc,))
         """
-
+        # if there are multiple streams
         if isinstance(nds[0], tuple):
             names, docs = list(zzip(*nds))
             if len(set(names)) > 1:
                 raise RuntimeError('Misaligned Streams')
             name = names[0]
+            newdocs = list()
+            for doc in docs:
+                # for case of ((name, ({}, {})), (name, ({}, {})))
+                if isinstance(doc, tuple):
+                    newdocs.extend(doc)
+                else:
+                    newdocs.append(doc)
+
+            docs = tuple(newdocs)
+
+        # if only one stream
         else:
             names, docs = nds
```

Comment:
this might not work for the case `('start', ({}, {}))`. (as far as I can see from reading) See comment in tests. We should test this.
(might need to check (`if len(docs) ==1` etc)